projects/packages/my-jetpack/src/products/class-product.php

```diff
@@ -443,21 +445,41 @@ public static function get_status() {
 			$status = 'active';
 			// We only consider missing site & user connection an error when the Product is active.
 			if ( static::$requires_site_connection && ! ( new Connection_Manager() )->is_connected() ) {
-				$status = 'error';
+				// Site has never been connected before
+				if ( ! \Jetpack_Options::get_option( 'id' ) ) {
+					$status = 'needs_first_site_connection';
+				} else {
+					$status = 'site_connection_error';
+				}
 			} elseif ( static::$requires_user_connection && ! ( new Connection_Manager() )->has_connected_owner() ) {
-				$status = 'error';
+				$status = 'user_connection_error';
 			} elseif ( static::is_upgradable() ) {
-				// Upgradable plans should ignore whether or not they have the required plan.
 				$status = 'can_upgrade';
-			} elseif ( ! static::has_required_plan() ) { // We need needs_purchase here as well because some products we consider active without the required plan.
-				if ( static::has_trial_support() ) {
-					$status = 'needs_purchase_or_free';
-				} else {
+			}
+			// Check specifically for inactive modules, which will prevent a product from being active
+		} elseif ( property_exists( static::class, 'module_name' ) && static::$module_name && ! static::is_module_active() ) {
+			$status = 'module_inactive';
+			// If there is not a plan associated with the disabled module, encourage a plan first
+			// Getting a plan set up should help resolve any connection issues
+			// However if the standalone plugin for this product is active, then we will defer to showing errors that prevent the module from being active
+			// This is because if a standalone plugin is installed, we expect the product to not show as "inactive" on My Jetpack
+			if ( ! static::has_any_plan_for_product() && static::$has_standalone_plugin && ! self::is_plugin_active() ) {
+				$status = 'needs_purchase_or_free';
+				if ( ! static::$has_free_offering ) {
 					$status = 'needs_purchase';
 				}
+			} elseif ( static::$requires_site_connection && ! ( new Connection_Manager() )->is_connected() ) {
+				// Site has never been connected before
+				if ( ! \Jetpack_Options::get_option( 'id' ) ) {
+					$status = 'needs_first_site_connection';
+				} else {
+					$status = 'site_connection_error';
+				}
```

Comment:
I see this exact conditional above in the previous if statement. Not sure its necessary since it's only used twice but we could potentially introduce some reusable variable or function for this check

Reply:
Might not be needed at this point - I think I'll leave these duplicated instances for now. Thanks for the suggestion though!